### PR TITLE
Added code of conduct

### LINKED ISSUE #2135: code of conduct
here's a rough draft:

---
# Mocha Code of Conduct

_This code of conduct is applicable to issues and pull requests on Mocha's GitHub repository, and its Gitter chat room._
1.  Be nice; don't attack or threaten anyone.  
   - If you _aren't_ nice, expect someone to let you know.
   - ...and if you let somebody know, be nice when you do it.
2.  It is difficult to convey tone or attitude in GitHub issues.  Be clear, and _slow down_.
3.  If you're new to GitHub or OSS, look around for resources on etiquette and how to get started--these will help you!
4.  Read `CONTRIBUTING.md` before you contribute.
5.  Mocha is free; please temper your expectations.

---

that's all I got.

### PR DESCRIPTION
**Pull request by kunagpal**
Resolves #2135

### CODE REVIEW

**Comment by boneskull** [comment]
Also, you can add `[ci skip]` to end of your commit message to avoid a build.

**Comment by kunagpal** [comment]
@boneskull Using `[ci skip]` makes sense, but I've noticed that it causes the check status section to be stuck in limbo.

**Comment by kunagpal** [comment]
@boneskull All up to date :)

**Comment by boneskull** [comment]
@kunagpal I think the hung builds you mention are the product of some other weird badness

File: .github/CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# Mocha code of conduct
```
Comment (boneskull) [inline .github/CODE_OF_CONDUCT.md]: please capitalize as `Mocha Code of Conduct`

**Comment by kunagpal** [comment]
@boneskull Figured as much, the latest commit skips the builds altogether.

File: .github/CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+Like the technical community as a whole, the various JS Foundation teams and communities are made up of a mixture of
```
Comment (boneskull) [inline .github/CODE_OF_CONDUCT.md]: I think we can remove lines 3 through 14.

Comment (kunagpal) [inline .github/CODE_OF_CONDUCT.md]: Done.

**Pull request merged**